Userland/Libraries/LibJS/Runtime/ExecutionContext.cpp

```diff
@@ -13,9 +13,42 @@
 
 namespace JS {
 
+class ExecutionContextAllocator {
+public:
+    NonnullOwnPtr<ExecutionContext> allocate(Heap& heap)
+    {
+        if (m_execution_contexts.is_empty())
+            return adopt_own(*new ExecutionContext(heap));
+        return m_execution_contexts.take_last();
+    }
+    void deallocate(ExecutionContext* ec)
+    {
+        if (m_destroyed)
+            return;
+        memset(ec, 0, sizeof(ExecutionContext));
+        m_execution_contexts.append(adopt_own(*ec));
+    }
+
+    ~ExecutionContextAllocator()
+    {
+        m_destroyed = true;
+    }
+
+private:
+    Vector<NonnullOwnPtr<ExecutionContext>> m_execution_contexts;
+    bool m_destroyed { false };
+};
+
+static ExecutionContextAllocator s_execution_context_allocator;
```

Comment:
Maybe this can be `static NeverDestroyed<ExecutionContextAllocator>` and we don't need the `m_destroyed` stuff.